### ISSUE TITLE
Error in importing VGG16NOTOP model using helper functions

### ISSUE THREAD

**Issue by asmehra95** [deeplearning4j/deeplearning4j#3099]
I tried to load VGGNoTop by using following helper function but it gave the following error:

`org.deeplearning4j.exception.DL4JInvalidConfigException: Invalid configuration for layer (idx=-1, name=block1_conv1, type=ConvolutionLayer) for height dimension:  Invalid input configuration for kernel height. Require 0 < kH <= inHeight + 2*padH; got (kH=3, inHeight=0, padH=0)
Input type = InputTypeConvolutional(h=0,w=0,d=3), kernel = [3, 3], strides = [1, 1], padding = [0, 0], layer size (output depth) = 64, convolution mode = Same
        at org.deeplearning4j.nn.conf.layers.InputTypeUtil.getOutputTypeCnnLayers(InputTypeUtil.java:54)
        at org.deeplearning4j.nn.conf.layers.ConvolutionLayer.getOutputType(ConvolutionLayer.java:95)
        at org.deeplearning4j.nn.modelimport.keras.layers.KerasConvolution.getOutputType(KerasConvolution.java:89)
        at org.deeplearning4j.nn.modelimport.keras.KerasModel.helperInferOutputTypes(KerasModel.java:286)
        at org.deeplearning4j.nn.modelimport.keras.KerasModel.<init>(KerasModel.java:171)
        at org.deeplearning4j.nn.modelimport.keras.KerasModel.<init>(KerasModel.java:97)
        at org.deeplearning4j.nn.modelimport.keras.KerasModel$ModelBuilder.buildModel(KerasModel.java:576)
        at org.deeplearning4j.nn.modelimport.keras.KerasModelImport.importKerasModelAndWeights(KerasModelImport.java:177)
        at org.deeplearning4j.nn.modelimport.keras.trainedmodels.TrainedModelHelper.loadModel(TrainedModelHelper.java:70)
        at org.apache.tika.parser.recognition.dl4j.DL4JImageRecogniser.initialize(DL4JImageRecogniser.java:119)
        ... 10 more`

i used the following code to load it.

`TrainedModelHelper helper;
                   if(modelType.equals("VGG16NOTOP"))
                    {
                    	helper = new TrainedModelHelper(TrainedModels.VGG16NOTOP);
                    }else if(modelType.equals("VGG16"))
                    {
                    	helper = new TrainedModelHelper(TrainedModels.VGG16);
                    }else
                    {
                    	throw new TikaConfigException("Unknown or unsupported model");
                    }
                    model = helper.loadModel();
    				/*model= KerasModelImport.importKerasModelAndWeights(MODEL_DIR+File.separator+"VGG16NoTop.json",MODEL_DIR+File.separator+"vgg16_weights_th_dim_ordering_th_kernels_notop",false);*/
    				LOG.info("Saving the Loaded model for future use. Saved models are more optimised to consume less resource.");
    				ModelSerializer.writeModel(model,locationToSave,true);`

VGG16 model loads perfectly but not VGG16NOTOP

**Comment by turambar** [comment]
@asmehra95 your error message tells you precisely what your problem is: the "input height" to your convolutional layer is 0. This is because the "no top" version of the Keras VGG16 architecture does not define its input size (`batch_input_shape=[None, None, None, None]`) but DL4J requires you to define an input size.

Possible solutions:
* import the full Keras VGG16 model and use DL4J's brand new [transfer learning API](https://github.com/deeplearning4j/deeplearning4j/tree/master/deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/transferlearning) to remove/replace the top.
* modify Keras model import to allow user to set input type at import time. This is on our radar, but we're not sure when we'll get to it. We'll gratefully accept a PR if you feel so inclined, though!

Thoughts @eraly?

**Comment by maxpumperla** [comment]
@asmehra95 have you had the chance to experiment with the transfer learning API @turambar was mentioning?

**Comment by maxpumperla** [comment]
@asmehra95 @turambar we went for solution 2 and allow optional input shape now, see e.g.

https://github.com/deeplearning4j/deeplearning4j/blob/37cef78cd2f221db5aba1668f21e3db6e3994b9d/deeplearning4j-modelimport/src/test/java/org/deeplearning4j/nn/modelimport/keras/e2e/KerasModelEndToEndTest.java#L333-L337
